CRM/Core/BAO/SchemaHandler.php

```diff
@@ -589,6 +589,10 @@ public static function checkIfFieldExists($tableName, $columnName, $i18nRewrite
    * @return bool TRUE if FK is found
    */
   public static function checkFKExists(string $table_name, string $constraint_name): bool {
+    if (!isset(\Civi::$statics['CRM_Core_DAO']['init'])) {
+      // This could get called early during installation.
+      return FALSE;
+    }
```

Comment:
That's funny, the installer is just getting ready to execute some sql to create tables, so you'd think the db would be available.